core/lib/multivm/src/versions/vm_fast/vm.rs

```diff
@@ -503,66 +499,67 @@ impl<S: ReadStorage> VmInterface for Vm<S> {
             // Move the pointer to the next transaction
             self.bootloader_state.move_tx_to_execute_pointer();
             track_refunds = true;
-            // Create a snapshot to roll back the bootloader call frame on halt.
-            self.make_snapshot();
```

Comment:
What is the reason for removing this snapshot and the rollbacks further in this function?

Reply:
I've introduced this additional snapshot in [my previous PR](https://github.com/matter-labs/zksync-era/pull/2378/files), and it looks redundant now that I understand VM function better 🙃 The only thing this snapshot did was zeroing the VM outputs before a rollback of the original snapshot (which would occur later), and this is easy to reproduce without taking a snapshot. Another option would be to ignore the VM outputs on the caller side, but it looks more complex to implement / maintain.

Reply:
Is the actual snapshotting done in the sequencer now?

Reply:
It was performed in the sequencer previously as well, but it's now performed *entirely* there, yes.